gulpfile.js

```diff
@@ -0,0 +1,35 @@
+const gulp = require('gulp');
+const gutil = require('gulp-util');
+const typescript = require('gulp-typescript');
+const sourcemaps = require('gulp-sourcemaps');
+const tsProject = typescript.createProject('tsconfig.json');
+
+const OPTIONS = {
+    header: true,
+    files: {
+        typescript: [ 'lib/**/*.ts' ],
+        declarations: [ 'lib/**/*.d.ts' ]
+    },
+    base: 'lib'
+}
+
+// Compile the TS sources
+gulp.task('typescript', () => {
+    gulp.src(OPTIONS.files.typescript)
+        .pipe(sourcemaps.init())
+        .pipe(tsProject()).on('error', gutil.log)
+        .pipe(sourcemaps.write('./', {
+			includeContent: false,
+            sourceRoot: '../lib'
```

Comment:
Related to my comment out the build output dir, the typescript compiler can do sourcemaps for you too. Do we need to do this separately ourselves? You should be able to configure internal/external, location, source roots and content inclusion all from normal tsconfig params.